pkg/storage/db.go

```diff
@@ -120,6 +122,61 @@ func (db *DB) Init(ctx context.Context, postgresConfig *PostgresConfig) error {
 	return initerr
 }
 
+// GeneratePartition finds the latest partition, and generate the next partition based on the previous partions time range
+func (db *DB) GeneratePartition(ctx context.Context) error {
+	stmt := `SELECT tablename FROM pg_catalog.pg_tables WHERE tablename LIKE 'aws_events_ips_hostnames_%' ORDER BY tablename DESC LIMIT 1`
+	row := db.sqldb.QueryRowContext(ctx, stmt)
+	var latest string
+	err := row.Scan(&latest)
+	switch err {
+	case nil:
+	case sql.ErrNoRows:
+		now := db.now
+		if now == nil {
```

Comment:
Opinion: You don't have to do this here if you want to make use of the `sync.Once` to set the value.